packages/mds-transactions-service/@types/index.ts

```diff
@@ -1,27 +1,134 @@
+  receipt_id: UUID
+  timestamp: Timestamp // could be any time before the Transaction 'created'
+  origin_url: URL // where can I go to dig into the details of the receipt, given this receipt_id?
+  receipt_details: Object // JSON blob with free-form supporting evidence, DO NOT INCLUDE PII
 }
 
+export interface TransactionDomainModel {
+  transaction_id: UUID
+  provider_id: UUID
+  device_id: Nullable<UUID> // optional
+  timestamp: Timestamp
+  fee_type: FEE_TYPE
+  amount: Number // pennies
+  receipt: ReceiptDomainModel // is this how you specify a JSON blob?
+}
 export type TransactionDomainCreateModel = DomainModelCreate<TransactionDomainModel>
 
+export const TRANSACTION_OPERATION_TYPES = [
+  'transaction_posted',
+  'invoice_generated',
+  'dispute_requested',
+  'dispute_approved',
+  'dispute_declined',
+  'dispute_canceled'
+] as const
+
+export type TRANSACTION_OPERATION_TYPE = typeof TRANSACTION_OPERATION_TYPES[number]
+
+export interface TransactionOperationDomainModel {
+  operation_id: UUID
+  transaction_id: UUID
+  // when was this change made
+  timestamp: Timestamp
+  operation_type: TRANSACTION_OPERATION_TYPE
+  // who made this change (TODO work out authorship representation; could be human, could be api, etc.)
+  author: string
+}
+export type TransactionOperationDomainCreateModel = DomainModelCreate<TransactionOperationDomainModel>
+
+export const TRANSACTION_STATUS_TYPES = [
+  'order_submitted',
+  'order_canceled',
+  'order_complete',
+  'order_incomplete'
+] as const
+
+export type TRANSACTION_STATUS_TYPE = typeof TRANSACTION_STATUS_TYPES[number]
+
+export interface TransactionStatusDomainModel {
+  status_id: UUID
+  transaction_id: UUID
+  // when was this change made
+  timestamp: Timestamp
+  status_type: TRANSACTION_OPERATION_TYPE
+  // who made this change (TODO work out authorship representation; could be human, could be api, etc.)
+  author: string
+}
+export type TransactionStatusDomainCreateModel = DomainModelCreate<TransactionStatusDomainModel>
+
 export interface TransactionService {
-  createTransactions: (transactions: TransactionDomainCreateModel[]) => TransactionDomainModel[]
+  // single
   createTransaction: (transaction: TransactionDomainCreateModel) => TransactionDomainModel
+  // bulk
+  createTransactions: (transactions: TransactionDomainCreateModel[]) => TransactionDomainModel[]
+
+  // read-back bulk TODO search criteria
   getTransactions: () => TransactionDomainModel[]
-  getTransaction: (name: string) => TransactionDomainModel
-  updateTransaction: (transaction: TransactionDomainModel) => TransactionDomainModel
-  deleteTransaction: (name: string) => TransactionDomainModel['name']
+  // read back single
+  getTransaction: (transaction_id: TransactionDomainModel['transaction_id']) => TransactionDomainModel
+
+  // create an 'operation', e.g. for dispute-handling, etc.
+  addTransactionOperation: (status: TransactionOperationDomainCreateModel) => TransactionOperationDomainModel
```

Comment:
Should param be called `operation` rather than `status`?

Reply:
yes